### PR TITLE
Introduced Handler pattern for subscription modification logic

### PR DESCRIPTION
**Pull request by nathandmoore**
For BroadleafCommerce/MicroPM#5130

### CODE REVIEW

**Review by ckittrell — APPROVED** [review]
A few details to cleanup, but looks good!

File: services/src/main/java/com/broadleafcommerce/subscriptionoperation/service/modification/AbstractModifySubscriptionHandler.java [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+     *        modification to make.
+     * @param contextInfo the current sandbox and multitenant context
+     * @throws InsufficientSubscriptionAccessException if the user does not have access to the
+     *         subscription
+     */
+    protected void validateUserAccessToSubscription(
+            @lombok.NonNull ModifySubscriptionRequest request,
+            @Nullable ContextInfo contextInfo) {
+        // TODO Will likely need components like AuthenticationVendorPrivilegesUtility &
+        // AuthenticationVendorPrivilegesSummary to validate permissions for each subscription
```
[the remark this answers is not in the document]
Comment (nathandmoore) [reply services/src/main/java/com/broadleafcommerce/subscriptionoperation/service/modification/AbstractModifySubscriptionHandler.java:100]: I don't know. I copied this from what Sunny had elsewhere. https://github.com/BroadleafCommerce/SubscriptionOperationServices/pull/13/files#diff-04b97df738a275de00e86e72b749ff536b60b0fec6bd22c1133e98f9af080c5fL178

File: services/src/main/java/com/broadleafcommerce/subscriptionoperation/web/domain/ModifySubscriptionRequest.java [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+     */
+    private String reason;
+
+    /**
+     * TODO
+     */
+    private boolean immediateCancellation = false;
+
+    /** TODO */
+    private Subscription newSubscription;
```
[the remark this answers is not in the document]
Comment (nathandmoore) [reply services/src/main/java/com/broadleafcommerce/subscriptionoperation/web/domain/ModifySubscriptionRequest.java:77]: Ya, I don't know if those will stick around. I copied them from the existing placeholder request DTOs as is and figured we'd revisit when the design for upgrade, downgrade, and cancellation were more concrete.

[the remark this answers is not in the document]
Comment (ckittrell) [reply services/src/main/java/com/broadleafcommerce/subscriptionoperation/web/domain/ModifySubscriptionRequest.java:77]: Got it. I'm fine with leaving as-is then.

File: services/src/main/java/com/broadleafcommerce/subscriptionoperation/service/modification/AbstractModifySubscriptionHandler.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (ckittrell) [reply services/src/main/java/com/broadleafcommerce/subscriptionoperation/service/modification/AbstractModifySubscriptionHandler.java:100]: I'm 95% sure that I know what it's referring to. Planning to leave it alone for now.

**Pull request merged**